src/components/KCodeBlock/KCodeBlock.vue

```diff
@@ -213,23 +164,87 @@
 
         <slot name="secondary-actions" />
       </div>
+
+      <!--
+        Instead of using <pre> directly, the <component is="pre"> here acts as a workaround for
+        a potential bug of Vue itself. Because the <Virtualizer> fails to render its scoped slots
+        when it's wrapped inside a <pre> element.
+      -->
+      <!-- eslint-disable-next-line vue/require-component-is -->
+      <component
+        is="pre"
+        v-if="isShowingFilteredCode || hasRenderedFilteredCode"
+        v-show="isShowingFilteredCode"
+        class="filtered-code-block"
+        data-testid="filtered-code-block"
+      >
+        <Virtualizer
+          v-if="!singleLine"
+          v-slot="{ item: line }"
+          v-bind="getVirtualizerProps(true)"
+        >
+          <a
+            :id="`${linePrefix}-L${line}`"
```

Comment:
suggestion: you utilize the `${linePrefix}-L${line}` in a few places in the loop. Could we extract this to a getter?